tasks/scripts/terraform-smoke

```diff
@@ -18,6 +39,13 @@ echo "$SSH_KEY" > keys/id_rsa
 chmod 0600 keys/id_rsa
 ssh-keygen -y -f keys/id_rsa > keys/id_rsa.pub
 
+if [[ $CONCOURSE_VERSION != "dev" ]]; then
+  cmp $CONCOURSE_VERSION "6.5.0"
```

Comment:
Actually, do you have to `set +e` before this? Does the command "fail" if the return code isn't 0?

Reply:
It looks like it: tried running the following script:

```sh
#!/bin/bash

set -e

cmp() {
  # strip pre-release (-rc) or build (+123) from semver
  v1=$(echo $1 | sed -e 's/[-+].*$//g')
  v2=$(echo $2 | sed -e 's/[-+].*$//g')

  if [[ $v1 == $v2 ]]; then
    return 0
  fi

  # use the magic of sort to figure out which version is smaller
  smaller=$(echo -e "$v1\n$v2" | sort -t "." -k 1,1 -k 2,2 -k 3,3 -g | head -n1)
  if [[ $smaller == $v1 ]]; then
    return 1
  fi
  return 2
}

if [[ 1 != 2 ]]; then
  cmp '1.2.3' '4.5.6'
  case $? in
    0) echo "=";;
    1) echo "<";;
    2) echo ">";;
  esac
fi
```

and it exited with error code 1 without `echo`ing anything. Adding `set +e` before the `cmp` and `set -e` after the `esac` allows it to work properly

Reply:
That's a good point, I think I'm gonna have it be based on output instead of exit codes, don't like the idea that we won't know if `sort` failed somehow.